setuptools/dist.py

```diff
@@ -272,6 +272,7 @@ class Distribution(_Distribution):
     }
 
     _patched_dist = None
+    namespace_packages: List[str]
```

Comment:
Personally, I would not add this one here, because setuptools augments many instance attributes dynamically via entry-points (it might be a lost cause to try to statically define all of them).

But if that is required to make the type checker happy, I think we should at least annotate something like the following.

```suggestion
    namespace_packages: List[str]  #: :meta private: DEPRECATED
```

Reply:
It's for https://github.com/pypa/setuptools/pull/4375/files#diff-6c888d2f35e4149ad56292334032e9a216d1ab2b32a4df3c791d864c42dfad04L522-R523

```suggestion
    # Used by setuptools/command/editable_wheel.py for legacy namespaces
    namespace_packages: List[str]  #: :meta private: DEPRECATED
```

Reply:
Alternatively, we can even make it statically depcrecated using Python 3.13 decoractor (backported in `typing_extensions`) with something like:
```suggestion
    if TYPE_CHECKING:
        # Used by setuptools/command/editable_wheel.py for legacy namespaces
        @property
        @deprecated("Some deprecated message about what to use instead.")
        def namespace_packages(self) -> List[str]: ...  #: :meta private: DEPRECATED
```

![image](https://github.com/pypa/setuptools/assets/1350584/4a65e7b5-f530-4b16-b99d-4160c229940f)

Reply:
Thank you very much @Avasam. The last change should be enough (always good to avoid the hassle of conditionally importing `typing_extensions`).

Reply:
Yeah and it results in a weird non-runtime property.